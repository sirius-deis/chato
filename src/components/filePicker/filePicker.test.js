import { render, screen, fireEvent } from "@testing-library/react";
import FilePicker from "./filePicker";

const icons = "";

describe("FilePicker component", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <FilePicker size={3} icon={icons}></FilePicker>
    );
    expect(container).toMatchSnapshot();
  });
  it("should render a file input element", () => {
    render(<FilePicker></FilePicker>);
    expect(screen.getByTestId("file")).toBeInTheDocument();
  });
  it("should render an icon", () => {
    render(<FilePicker></FilePicker>);
    expect(screen.getByTestId("file-icon")).toBeInTheDocument();
  });
  it("should fire a change event when a file is selected", () => {
    const fn = jest.fn();
    render(<FilePicker onChange={fn}></FilePicker>);
    const file = new File(["(⌐□_□)"], "chucknorris.png", { type: "image/png" });
    fireEvent.change(screen.getByTestId("file"), {
      target: { files: [file] },
    });
    expect(fn).toHaveBeenCalled();
  });
});
