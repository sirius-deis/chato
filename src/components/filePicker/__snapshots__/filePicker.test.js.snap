// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePicker component should match snapshot 1`] = `
<div>
  <label
    class="sc-aXZVg dhWhfl"
  >
    <input
      class="sc-gEvEer iuhuvm"
      data-testid="file"
      type="file"
    />
    <div
      class="sc-eqUAAy gCcMHU"
      data-testid="file-icon"
      size="3"
    />
  </label>
</div>
`;
